### ISSUE TITLE
"Substring must be compatible" issue

### ISSUE THREAD

**Issue by dfeldmanaf** [shiftonelabs/laravel-sqs-fifo-queue#23]
Declaration of ShiftOneLabs\LaravelSqsFifoQueue\Support\Str::substr($string, $start, $length = null) must be compatible with Illuminate\Support\Str::substr($string, $start, $length = null, $encoding = 'UTF-8') {"userId":1,"exception":"[object] (Symfony\\Component\\ErrorHandler\\Error\\FatalError(code: 0): Declaration of ShiftOneLabs\\LaravelSqsFifoQueue\\Support\\Str::substr($string, $start, $length = null) must be compatible with Illuminate\\Support\\Str::substr($string, $start, $length = null, $encoding = 'UTF-8') at /Users/drew/product/vendor/shiftonelabs/laravel-sqs-fifo-queue/src/Support/Str.php:47)

I was able to fix the error by adding encode as a param to this function in the str.php file

```
/**
     * Returns the portion of string specified by the start and length parameters.
     *
     * The substr method wasn't added to the Str class until Laravel 5.1.
     * Add the implementation here to support older versions of Laravel.
     *
     * @param  string  $string
     * @param  int  $start
     * @param  int|null  $length
     *
     * @return string
     */
    public static function substr($string, $start, $length = null, $encoding = 'UTF-8')
    {
        return mb_substr($string, $start, $length, $encoding);
    }

```
That being said, I know it's not a real fix. I'm a fairly new developer, can someone help me figure out what this solution is?

**Comment by dfeldmanaf** [comment]
I should clarify these are in my composer file

"laravel/framework": "9.44.0",
"shiftonelabs/laravel-sqs-fifo-queue": "^2.0",

When laravel framework was 9.43.0, it worked fine. As soon as it updated to 9.44.0 i got this error.

**Comment by patrickcarlohickman** [comment]
@dfeldmanaf,

This was recently fixed last week with the release of the 3.0.0 tag. Please update your composer dependency to use "^3.0" instead of "^2.0".

Hope this helps.

Thanks!
Patrick
